sources/Algo/include/Algo.h

```diff
@@ -182,23 +200,31 @@ struct GeneratorDefinition {
 class GeneratorDefinitionAlgorithm : public NodeAlgorithm {
  public:
   using Generators = std::vector<GeneratorDefinition>;  ///< alias for list of generators
+  using BusId = std::string;                            ///< alias for bus id
+  using GenId = std::string;                            ///< alias for generator id
+  using BusGenMap = std::unordered_map<BusId, GenId>;   ///< alias for map of bus id to generator id
 
   /**
    * @brief Constructor
    *
    * @param gens generators list to update
+   * @param busMap mapping of busId and the number of generators that regulates them
    * @param infinitereactivelimits parameter to determine if infinite reactive limits are used
    * @param serviceManager dynawo service manager in order to use Dynawo extra algorithms
+   * @param buses map of bus ids to a generator that regulates them
    */
-  GeneratorDefinitionAlgorithm(Generators& gens, bool infinitereactivelimits, const boost::shared_ptr<DYN::ServiceManagerInterface>& serviceManager);
+  GeneratorDefinitionAlgorithm(Generators& gens, const inputs::NetworkManager::BusMapRegulating& busMap, bool infinitereactivelimits,
+                               const boost::shared_ptr<DYN::ServiceManagerInterface>& serviceManager, BusGenMap& buses);
```

Comment:
logic for outputs / inputs separation. Put all outputs on one side of the arguments